### PR TITLE
add empty string check for tokenTo and tokenFrom

### LINKED ISSUE #1341: testpoolswap returns estimate when required fields are missing
#### What happened:
`testpoolswap` returns what could be mistaken for a valid estimate when the `tokenFrom` and `tokenTo` fields are missing

#### What you expected to happen:
I expect `testpoolswap` to fail with an error when any required fields (according to the documentation under `defi-cli help testpoolswap`) are missing

#### How to reproduce it (as minimally and precisely as possible):
You can submit the following request:
`{"jsonrpc":"1.0","method":"testpoolswap","params":[{"amountFrom":10.0,"tokenFrom":"","tokenTo":"","from":"dGHXU1ZCSmq7dQvezb9ZdENJCoDgiQBAb1","to":"dGHXU1ZCSmq7dQvezb9ZdENJCoDgiQBAb1"},"auto"]}`

The result I got:
`{"result":"9.96001916@0","error":null,"id":null}`

#### What are your environment parameters:
Running version 2.8.1 on Windows.

My config:
rpcuser=XXX
rpcpassword=XXX
rpcport=8332
prune=1200
listen=0

zmqpubrawblock=tcp://127.0.0.1:29000
zmqpubrawtx=tcp://127.0.0.1:29000
zmqpubhashtx=tcp://127.0.0.1:29000
zmqpubhashblock=tcp://127.0.0.1:29000

### PR DESCRIPTION
**Pull request by shohamc1**
#### What kind of PR is this?:

/kind fix

#### What this PR does / why we need it:
Checks for an empty `tokenTo` and `tokenFrom` in `testpoolswap`. Previously the empty string evaluated as DFI on both fields, and the RPC incorrectly returned the DFI-DFI composite swap output.

#### Which issue(s) does this PR fixes?:

Fixes #1341

### CODE REVIEW

**Review by Jouzo — DISMISSED** [review]

**Review by mambisi — APPROVED** [review]

File: src/masternodes/rpc_poolpair.cpp [hunk 1]
```diff
@@ -134,6 +134,13 @@ UniValue poolPathsToJSON(std::vector<std::vector<DCT_ID>> & poolPaths) {
 void CheckAndFillPoolSwapMessage(const JSONRPCRequest& request, CPoolSwapMessage &poolSwapMsg) {
     std::string tokenFrom, tokenTo;
     UniValue metadataObj = request.params[0].get_obj();
+
+    if (metadataObj["tokenFrom"].getValStr() == "") {
```
Comment (mambisi) [inline src/masternodes/rpc_poolpair.cpp:138]: `metadataObj["tokenFrom"].empty()`

File: src/masternodes/rpc_poolpair.cpp [hunk 2]
```diff
@@ -134,6 +134,13 @@ UniValue poolPathsToJSON(std::vector<std::vector<DCT_ID>> & poolPaths) {
...
+        throw JSONRPCError(RPC_INVALID_PARAMETER, "tokenFrom is empty");
+    }
+    if (metadataObj["tokenTo"].getValStr() == "") {
```
Comment (mambisi) [inline src/masternodes/rpc_poolpair.cpp:141]: `metadataObj["tokenTo"].empty()`

**Review by mambisi — COMMENTED** [review]
Can use `UniValue` `::empty` function

File: src/masternodes/rpc_poolpair.cpp (hunk 1 above)
Comment (shohamc1) [reply src/masternodes/rpc_poolpair.cpp:138]: I did try this, but that throws the error when `tokenFrom` is not empty.

Comment (Bushstar) [reply src/masternodes/rpc_poolpair.cpp:138]: `if (metadataObj["tokenFrom"].getValStr().empty())`

**Review by Jouzo — APPROVED** [review]

**Review by hidiego — APPROVED** [review]

**Comment by prasannavl** [comment]
Awaiting rebase to master. cc/ @shohamc1

**Comment by shohamc1** [comment]
Superseded by #1544

**Pull request closed without merging**